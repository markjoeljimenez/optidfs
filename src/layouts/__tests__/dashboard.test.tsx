import '@testing-library/jest-dom';
import { render, screen } from '../../scripts/testing/render';
import { mockNextUseRouter } from '../../scripts/testing/util/mockNextUseRouter';
import Dashboard from '../dashboard';

describe('Dashboard', () => {
	mockNextUseRouter({
		route: '/',
		pathname: '/test',
		query: '',
		asPath: ``,
	});

	it('renders dashboard successfully', () => {
		const dashboard = render(<Dashboard>{'test'}</Dashboard>);

		const nav = screen.getByTestId('nav');
		const header = screen.getByTestId('header');
		const mainContent = screen.getByTestId('main-content');

		expect(header).toBeInTheDocument();
		expect(nav).toBeInTheDocument();
		expect(mainContent).toHaveTextContent('test');
		expect(dashboard).toMatchSnapshot();
	});
});
