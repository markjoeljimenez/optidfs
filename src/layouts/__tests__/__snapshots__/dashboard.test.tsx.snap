// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders dashboard successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="md:flex md:min-h-screen bg-gray-100"
      >
        <nav
          class="px-6 py-4 md:py-8 border-r border-gray-200 font-bold flex flex-col bg-white"
          data-testid="nav"
        >
          <div
            class="relative flex justify-between items-center md:block md:min-w-nav"
          >
            <button
              class="md:hidden"
              type="button"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                >
                  <g
                    data-name="menu"
                  >
                    <rect
                      height="24"
                      opacity="0"
                      transform="rotate(180 12 12)"
                      width="24"
                    />
                    <rect
                      height="2"
                      rx=".95"
                      ry=".95"
                      width="18"
                      x="3"
                      y="11"
                    />
                    <rect
                      height="2"
                      rx=".95"
                      ry=".95"
                      width="18"
                      x="3"
                      y="16"
                    />
                    <rect
                      height="2"
                      rx=".95"
                      ry=".95"
                      width="18"
                      x="3"
                      y="6"
                    />
                  </g>
                </g>
              </svg>
              <span
                class="sr-only"
              >
                Menu
              </span>
            </button>
            <h1
              class="absolute ml-10 md:ml-0 md:relative md:text-3xl font-black whitespace-no-wrap"
            >
              Optidfs
            </h1>
          </div>
          <div
            class="md:max-h-none overflow-hidden transition-all duration-300 flex-1 flex flex-col justify-between space-y-6 text-gray-600 font-medium max-h-0"
          >
            <ul>
              <li
                class="mt-6"
              >
                <a
                  class="hover:bg-gray-200 block px-4 py-3 rounded-md"
                  href="/"
                >
                  <svg
                    class="h-5 w-5 absolute top-50 text-gray-400"
                    fill="none"
                    stroke="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11 4a2 2 0 114 0v1a1 1 0 001 1h3a1 1 0 011 1v3a1 1 0 01-1 1h-1a2 2 0 100 4h1a1 1 0 011 1v3a1 1 0 01-1 1h-3a1 1 0 01-1-1v-1a2 2 0 10-4 0v1a1 1 0 01-1 1H7a1 1 0 01-1-1v-3a1 1 0 00-1-1H4a2 2 0 110-4h1a1 1 0 001-1V7a1 1 0 011-1h3a1 1 0 001-1V4z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="pl-8"
                  >
                    Optimize
                  </span>
                </a>
              </li>
            </ul>
            <div
              class="flex items-center flex-col space-y-6 text-center"
            >
              <form
                action="https://www.paypal.com/cgi-bin/webscr"
                method="post"
                target="_top"
              >
                <input
                  name="cmd"
                  type="hidden"
                  value="_donations"
                />
                <input
                  name="business"
                  type="hidden"
                  value="DG67QWH46T6SC"
                />
                <input
                  name="item_name"
                  type="hidden"
                  value="Supporting the dev"
                />
                <input
                  name="currency_code"
                  type="hidden"
                  value="CAD"
                />
                <input
                  alt="Donate with PayPal button"
                  name="submit"
                  src="https://www.paypalobjects.com/en_US/i/btn/btn_donate_SM.gif"
                  title="PayPal - The safer, easier way to pay online!"
                  type="image"
                />
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Donate"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </form>
            </div>
          </div>
        </nav>
        <div
          class="md:flex md:flex-col w-full"
        >
          <header
            class="border-b border-gray-200 bg-white"
            data-testid="header"
          >
            <div
              class="mx-auto py-4 px-6 md:p-8 md:flex justify-between items-center"
            />
          </header>
          <main
            data-testid="main-content"
          >
            test
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="md:flex md:min-h-screen bg-gray-100"
    >
      <nav
        class="px-6 py-4 md:py-8 border-r border-gray-200 font-bold flex flex-col bg-white"
        data-testid="nav"
      >
        <div
          class="relative flex justify-between items-center md:block md:min-w-nav"
        >
          <button
            class="md:hidden"
            type="button"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
              >
                <g
                  data-name="menu"
                >
                  <rect
                    height="24"
                    opacity="0"
                    transform="rotate(180 12 12)"
                    width="24"
                  />
                  <rect
                    height="2"
                    rx=".95"
                    ry=".95"
                    width="18"
                    x="3"
                    y="11"
                  />
                  <rect
                    height="2"
                    rx=".95"
                    ry=".95"
                    width="18"
                    x="3"
                    y="16"
                  />
                  <rect
                    height="2"
                    rx=".95"
                    ry=".95"
                    width="18"
                    x="3"
                    y="6"
                  />
                </g>
              </g>
            </svg>
            <span
              class="sr-only"
            >
              Menu
            </span>
          </button>
          <h1
            class="absolute ml-10 md:ml-0 md:relative md:text-3xl font-black whitespace-no-wrap"
          >
            Optidfs
          </h1>
        </div>
        <div
          class="md:max-h-none overflow-hidden transition-all duration-300 flex-1 flex flex-col justify-between space-y-6 text-gray-600 font-medium max-h-0"
        >
          <ul>
            <li
              class="mt-6"
            >
              <a
                class="hover:bg-gray-200 block px-4 py-3 rounded-md"
                href="/"
              >
                <svg
                  class="h-5 w-5 absolute top-50 text-gray-400"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 4a2 2 0 114 0v1a1 1 0 001 1h3a1 1 0 011 1v3a1 1 0 01-1 1h-1a2 2 0 100 4h1a1 1 0 011 1v3a1 1 0 01-1 1h-3a1 1 0 01-1-1v-1a2 2 0 10-4 0v1a1 1 0 01-1 1H7a1 1 0 01-1-1v-3a1 1 0 00-1-1H4a2 2 0 110-4h1a1 1 0 001-1V7a1 1 0 011-1h3a1 1 0 001-1V4z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <span
                  class="pl-8"
                >
                  Optimize
                </span>
              </a>
            </li>
          </ul>
          <div
            class="flex items-center flex-col space-y-6 text-center"
          >
            <form
              action="https://www.paypal.com/cgi-bin/webscr"
              method="post"
              target="_top"
            >
              <input
                name="cmd"
                type="hidden"
                value="_donations"
              />
              <input
                name="business"
                type="hidden"
                value="DG67QWH46T6SC"
              />
              <input
                name="item_name"
                type="hidden"
                value="Supporting the dev"
              />
              <input
                name="currency_code"
                type="hidden"
                value="CAD"
              />
              <input
                alt="Donate with PayPal button"
                name="submit"
                src="https://www.paypalobjects.com/en_US/i/btn/btn_donate_SM.gif"
                title="PayPal - The safer, easier way to pay online!"
                type="image"
              />
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271%27%20height=%271%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Donate"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </form>
          </div>
        </div>
      </nav>
      <div
        class="md:flex md:flex-col w-full"
      >
        <header
          class="border-b border-gray-200 bg-white"
          data-testid="header"
        >
          <div
            class="mx-auto py-4 px-6 md:p-8 md:flex justify-between items-center"
          />
        </header>
        <main
          data-testid="main-content"
        >
          test
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
